src/plant_species/species.repository.ts

```diff
@@ -0,0 +1,158 @@
+import { NewSpeciesDTO } from "./../shared/new-species-dto";
+import { HttpException, HttpStatus, Injectable } from "@nestjs/common";
+import { Prisma } from "@prisma/client";
+import { PrismaService } from "nestjs-prisma";
+import { UpdateSpeciesDTO } from "../shared/update-species-dto";
+
+@Injectable()
+export class SpeciesRepository {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async createSpecies(newSpeciesDTO: NewSpeciesDTO) {
+    try {
+      return await this.prisma.plantSpecies.create({
+        data: {
+          name: newSpeciesDTO.name,
+          OptimalCo2: newSpeciesDTO.optimalCo2
+            ? newSpeciesDTO.optimalCo2
+            : null,
+          optimalTemperature: newSpeciesDTO.optimalTemperature
+            ? newSpeciesDTO.optimalTemperature
+            : null,
+          optimalHumidity: newSpeciesDTO.optimalHumidity
+            ? newSpeciesDTO.optimalHumidity
+            : null,
```

Comment:
```suggestion
          OptimalCo2: newSpeciesDTO.optimalCo2,
          optimalTemperature: newSpeciesDTO.optimalTemperature,
          optimalHumidity: newSpeciesDTO.optimalHumidity,
```

Can't see why it would be necessary to do a ternary here.

Might wanna test it locally first :)